src/Bicep.LangServer/Handlers/BicepDecompileCommandHandler.cs

```diff
@@ -0,0 +1,311 @@
+
+            // Show disclaimer
+            Log(output, TemplateDecompiler.DecompilerDisclaimerMessage);
+
+            Log(output, $"Finished decompiling to {mainBicepPath}");
+
+            // Show main decompiled file            
+            if (clientCapabilitiesProvider.DoesClientSupportShowDocumentRequest())
+            {
+                await server.Window.ShowDocument(new() { TakeFocus = true, Uri = DocumentUri.File(mainBicepPath) });
+            }
+
+            return (
+                result: new BicepDecompileCommandResult(BicepDecompileCommandStatus.Success, output.ToString(), new Uri(mainBicepPath, UriKind.Absolute), null),
+                successTelemetry: BicepTelemetryEvent.DecompileSuccess(pathsToSave.Length, telemetryProperties.CountConflictingFiles, telemetryProperties.OverwriteFiles)
+                );
+        }
+
+        private async Task<(string outputFolder, PathToSave[] pathsToSave)?> DeterminateFinalPathsToSave(StringBuilder output, string outputFolder, PathToSave[] pathsToSave, string mainBicepPath, TelemetryProperties telemetryProperties)
+        {
+            pathsToSave = pathsToSave.ToArray(); // clone
+            PathToSave[] existingPaths = pathsToSave.Where(f => File.Exists(f.Path)).ToArray();
+            var singleFileDecompilation = pathsToSave.Length == 1;
+            bool overwriteFiles = false;
+
+            telemetryProperties.CountConflictingFiles = existingPaths.Length;
+
+            if (existingPaths.Any())
+            {
+                if (clientCapabilitiesProvider.DoesClientSupportShowMessageRequest())
+                {
+                    Debug.Assert(!singleFileDecompilation || existingPaths[0].Path == pathsToSave[0].Path);
+
+                    var overwriteAction = new MessageActionItem() { Title = singleFileDecompilation ? "Overwrite" : "Overwrite all" };
+                    var createCopyAction = new MessageActionItem() { Title = singleFileDecompilation ? "Create copy" : "New subfolder" };
+                    var cancelAction = new MessageActionItem() { Title = "Cancel" };
+                    var conflictFilesWithQuotes = string.Join(", ", existingPaths.Select(f => "\"" + Path.GetRelativePath(outputFolder, f.Path) + "\""));
+                    var message =
+                        singleFileDecompilation ?
+                        $"Output file already exists: {conflictFilesWithQuotes}" :
+                        $"There are multiple decompilation output files and the following already exist: {conflictFilesWithQuotes}";
+                    Log(output, message);
+
+                    var result = await server.Window.ShowMessageRequest(
+                        new()
+                        {
+                            Message = message,
+                            Actions = new(new[] { overwriteAction, createCopyAction, cancelAction })
+                        });
+
+                    if (result.Title == overwriteAction.Title)
```

Comment:
I'm getting an exception here when I try this branch out in a Codespaces instance:

```
TRACE: Decompiling /workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.json into Bicep...
TRACE: Building semantic model for file:///workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.bicep
TRACE: Building semantic model for file:///workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.bicep
TRACE: Building semantic model for file:///workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.bicep
TRACE: Building semantic model for file:///workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.bicep
TRACE: Bicep.LangServer Information: 0 : 
TRACE: Decompilation main output: /workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.bicep
TRACE: Bicep.LangServer Information: 0 : 
TRACE: Decompilation all files to save: /workspaces/bicep/src/Bicep.Decompiler.IntegrationTests/Files/Working/201-vm-copy-index-loops/main.bicep
TRACE: Bicep.LangServer Information: 0 : 
TRACE: Output file already exists: "main.bicep"
6:40:25 AM: Error: Internal Error - System.NullReferenceException: Object reference not set to an instance of an object.
   at Bicep.LanguageServer.Handlers.BicepDecompileCommandHandler.DeterminateFinalPathsToSave(StringBuilder output, String outputFolder, PathToSave[] pathsToSave, String mainBicepPath, TelemetryProperties telemetryProperties) in /workspaces/bicep/src/Bicep.LangServer/Handlers/BicepDecompileCommandHandler.cs:line 228
   at Bicep.LanguageServer.Handlers.BicepDecompileCommandHandler.Decompile(String jsonPath) in /workspaces/bicep/src/Bicep.LangServer/Handlers/BicepDecompileCommandHandler.cs:line 160
   at Bicep.LanguageServer.Handlers.BicepDecompileCommandHandler.<>c__DisplayClass9_0.<<Handle>b__0>d.MoveNext() in /workspaces/bicep/src/Bicep.LangServer/Handlers/BicepDecompileCommandHandler.cs:line 112
--- End of stack trace from previous location ---
   at Bicep.LanguageServer.Handlers.TelemetryAndErrorHandlingHelper`1.ExecuteWithTelemetryAndErrorHandling(Func`1 executeFunc) in /workspaces/bicep/src/Bicep.LangServer/Handlers/TelemetryAndErrorHandlingHelper.cs:line 46
   at OmniSharp.Extensions.LanguageServer.Server.Pipelines.SemanticTokensDeltaPipeline`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at OmniSharp.Extensions.LanguageServer.Server.Pipelines.ResolveCommandPipeline`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPreProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestPostProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at MediatR.Pipeline.RequestExceptionActionProcessorBehavior`2.Handle(TRequest request, CancellationToken cancellationToken, RequestHandlerDelegate`1 next)
   at OmniSharp.Extensions.JsonRpc.RequestRouterBase`1.<RouteRequest>g__InnerRoute|7_0(IServiceScopeFactory serviceScopeFactory, Request request, TDescriptor descriptor, Object params, CancellationToken token, ILogger logger)
   at OmniSharp.Extensions.JsonRpc.RequestRouterBase`1.RouteRequest(IRequestDescriptor`1 descriptors, Request request, CancellationToken token)
   at OmniSharp.Extensions.JsonRpc.DefaultRequestInvoker.<>c__DisplayClass10_0.<<RouteRequest>b__5>d.MoveNext()
```

Can confirm it's caused by `result` being null:
<img width="835" alt="image" src="https://user-images.githubusercontent.com/38542602/192958762-c85de752-f0ce-48c9-8738-4a9274630b7e.png">

Reply:
That code no longer exists, it's now in the vscode extension.